src/Engine/Contest/Solvers/ContestDijkstra.cpp

```diff
@@ -237,26 +240,34 @@ ContestDijkstra::AddEdges(const ScanTaskPoint origin,
        search */
     destination.SetPointIndex(first_finish_candidate);
 
+  const GeoPoint originGeo = GetPoint(origin).GetLocation();
   const unsigned weight = GetStageWeight(origin.GetStageNumber());
 
   bool previous_above = false;
   for (const ScanTaskPoint end(destination.GetStageNumber(), n_points);
        destination != end; destination.IncrementPointIndex()) {
     bool above = GetPoint(destination).GetIntegerAltitude() >= min_altitude;
 
-    if (above) {
-      const unsigned d = weight * CalcEdgeDistance(origin, destination);
-      Link(destination, origin, d);
-    } else if (previous_above) {
-      /* After excessive thinning, the exact TracePoint that matches
-         the required altitude difference may be gone, and the
-         calculated result becomes overly pessimistic.  This code path
-         makes it optimistic, by checking if the previous point
-         matches. */
-
-      /* TODO: interpolate the distance */
-      const unsigned d = weight * CalcEdgeDistance(origin, destination);
-      Link(destination, origin, d);
+    const unsigned distance = GetPoint(destination).DistanceTo(originGeo);
+
+    // Check if the distance is withing the minimum distance.
+    // Also allows zero distance legs, because if a minimum distance is set not
+    // all solutions will use all legs.
```

Comment:
This is rather obscure - can't we find a better, more reliable and easier-to-understand way to check if a leg is used at all?

Reply:
It's the least intrusive way I found, otherwise the `Dijkstra.hpp` needed significant changes to handle varying amount of stages. IMO this is fine, 0 length legs don't show up in the interface and work quite nicely.

Reply:
OK, agree, leave it like that for now. But (tiny nitpick) please use proper multi-line comment syntax for multi-line comments. `//` is a C++ extension explicitly designed for single-line comments.